### PR TITLE
Dropout

### PR DESCRIPTION
**Pull request by anderspapitto**
(most of this diff is the elman rnn that it's based on - the actual diff is just deleting the lockdowns

### CODE REVIEW

File: torch/onnx/symbolic.py [hunk 1]
```diff
@@ -534,6 +534,8 @@ def RNN_symbolic_builder(cell_type, *args, **kwargs):
         return LSTM_symbolic_builder(*args, **kwargs)
     elif cell_type == 'GRU':
         return GRU_symbolic_builder(*args, **kwargs)
+    elif cell_type.startswith('RNN_'):
```
Comment (houseroad) [inline torch/onnx/symbolic.py]: Here the meaning of "RNN_" should be the in-place version. Probably you should handle both RNN_ and RNN.

**Comment by anderspapitto** [comment]
@houseroad that's not part of this diff, that's part of the underlying https://github.com/pytorch/pytorch/pull/4613

This diff Is only the part that unlocks dropout

**Comment by ezyang** [comment]
Lint failed.

**Comment by ezyang** [comment]
Is this legit?  You have a different network if you don't have dropout, and that's effectively what this patch is doing.

Perhaps what you actually wanted to do is ignore dropout in inference mode, because in that case no dropout occurs. That's OK (test for THAT case), and error if someone tries to export in training mode. An example where we do this test already is FeatureDropout.

File: torch/onnx/symbolic.py [hunk 2]
```diff
@@ -543,11 +545,43 @@ def reform_weights(g, w, n, intervals):
     return g.op('Concat', *slices, axis_i=0)
 
 
+def Elman_RNN_symbolic_builder(nonlinearity, input_size, hidden_size, num_layers, batch_first, dropout, bidirectional, **kwargs):
```
Comment (ezyang) [inline torch/onnx/symbolic.py]: I understand that `kwargs` shows up in the other RNN implementations, but is there a reason it is done this way? It seems safer to explicitly list the arguments, so that if someone adds a new argument they will notice the symbolic needs to be updated.
